### PR TITLE
Relative paths for system/jvm properties

### PR DESCRIPTION
**Pull request by jrhee17**
Motivation:

We can observe absolute paths in:
- `--javaagent`
- `-XX:HeapDumpPath`
- `com.linecorp.armeria.blockhound.reportFile`
https://ge.armeria.dev/s/otz7vag6rsats/custom-values

Modifying to use relative paths would fix the issue easily.

Verifiable by 
- https://github.com/line/armeria/actions/runs/5611962575?pr=5055
- https://github.com/line/armeria/pull/5055/commits/1216bf709c172c41c8049141ca65b27139c1b4a2
  - See `reports-JVM-self-hosted-19` to ensure that files are created in the correct location

Screenshot 2023-07-21 at 12 50 25 AM

Modifications:

- Use relative paths for system/jvm properties

Result:

- More consistent inputs across gradle builds

Note: Although gradle suggests using the [following](https://docs.gradle.org/current/userguide/caching_java_projects.html#dealing_with_file_paths), I've decided against it due to an unnecessary dependency on the `build` directory
- https://github.com/line/armeria/actions/runs/5607879620/job/15192453765

### CODE REVIEW

File: .github/workflows/actions_build.yml [hunk 1]
```diff
@@ -166,12 +166,12 @@ jobs:
         if: failure()
         run: |
           find . '(' \
-          -name 'java_pid*.hprof' -or \
+          -name '*/java_pid*.hprof' -or \
```
Comment (jrhee17) [inline .github/workflows/actions_build.yml:169]:
```suggestion
          -name 'java_pid*.hprof' -or \
```

File: .github/workflows/actions_build.yml [hunk 2]
```diff
@@ -166,12 +166,12 @@ jobs:
...
           -name 'hs_err_*.log' -or \
           -path '*/build/reports/tests' -or \
           -path '*/build/test-results' -or \
           -path '*/javadoc.options' -or \
-          -name 'blockhound.log' ')' \
+          -name '*/blockhound.log' ')' \
```
Comment (jrhee17) [inline .github/workflows/actions_build.yml:174]:
```suggestion
          -name 'blockhound.log' ')' \
```

**Review by trustin — APPROVED** [review]
:+1:

**Review by minwoox — APPROVED** [review]
👍 👍 👍

**Review by ikhoon — APPROVED** [review]
👍👍

**Pull request merged**